frontend/src/hooks/useActions.js

```diff
@@ -563,4 +504,39 @@ const promptLoadFile = (parse, onData, errorMessage = 'Failed to parse file') =>
   input.click()
 }
 
+// Takes in the IDs of states, comments, and transitions
+// Parameters also include  the current project and whether a template is being created
+// Outputs a CopyData to be copied or Template object to be created into a template
+const selectionToCopyTemplate = (stateIds, commentIds, transitionIds, project, isTemplate) => {
```

Comment:
This method takes a selection and makes it into a CopyData or Template for the COPY or CREATE_TEMPLATE actions